src/pages/TransactionInfoPage.tsx

```diff
@@ -40,78 +43,55 @@ import TableBody from '@/components/Table/TableBody'
 import TableRow from '@/components/Table/TableRow'
 import Timestamp from '@/components/Timestamp'
 import TransactionIOList from '@/components/TransactionIOList'
-import { useGlobalContext } from '@/contexts/global'
-import useInterval from '@/hooks/useInterval'
-import { getAssetInfo } from '@/utils/assets'
 
 type ParamTypes = {
   id: string
 }
 
 const TransactionInfoPage = () => {
   const { id } = useParams<ParamTypes>()
-  const { client, networkType } = useGlobalContext()
-  const [txInfo, setTxInfo] = useState<explorer.Transaction>()
-  const [txBlock, setTxBlock] = useState<explorer.BlockEntryLite>()
-  const [txChain, setTxChain] = useState<explorer.PerChainHeight>()
-  const [txInfoStatus, setTxInfoStatus] = useState<number>()
-  const [txInfoError, setTxInfoError] = useState('')
-  const [loading, setLoading] = useState(true)
   const isAppVisible = usePageVisibility()
 
-  const getTxInfo = useCallback(async () => {
-    const fetchTransactionInfo = async () => {
-      if (!client || !id) return
-
-      setLoading(true)
-
-      try {
-        const data = await client.transactions.getTransactionsTransactionHash(id)
-        const tx = data as explorer.Transaction
-
-        if (tx) setTxInfo(tx)
-
-        if (!isTxConfirmed(tx)) return
-
-        const block = await client.blocks.getBlocksBlockHash(tx.blockHash)
-        const chainHeights = await client.infos.getInfosHeights()
-
-        setTxBlock(block)
+  const previousTransactionData = useRef<Transaction | undefined>()
+
+  const {
+    data: transactionData,
+    error: transactionInfoError,
+    isLoading: txInfoLoading
+  } = useQuery({
+    ...queries.transactions.detail(id || ''),
+    enabled: !!id,
+    refetchInterval:
+      isAppVisible && (!previousTransactionData.current || !isTxConfirmed(previousTransactionData.current))
+        ? 10000
+        : undefined
+  })
+
+  let txInfoError, txInfoErrorStatus
+
+  if (transactionInfoError) {
+    const e = transactionInfoError as APIError
+    txInfoError = e.error
+    txInfoErrorStatus = e.status
+  }
 
-        const chain = chainHeights.find(
-          (c: explorer.PerChainHeight) => c.chainFrom === block.chainFrom && c.chainTo === block.chainTo
-        )
+  previousTransactionData.current = transactionData as Transaction
 
-        setTxChain(chain)
-      } catch (e) {
-        console.error(e)
-        const { error, status } = e as APIError
+  const txInfo = transactionData as Transaction
 
-        setTxInfoStatus(status)
-        setTxInfoError(error.detail || error.message || 'Unknown error')
-      } finally {
-        setLoading(false)
-      }
-    }
+  const { data: txBlock } = useQuery({ ...queries.blocks.detail(txInfo?.blockHash), enabled: isTxConfirmed(txInfo) })
```

Comment:
Optional chaining is not needed

Reply:
It actually is. The problem is that I cast the `Transaction` type above (I should specify that this could be undefined as well). This casting is a relic of the previous approach, let me see if I can avoid it.